pertpy/tools/_distances/_distances.py

```diff
@@ -12,16 +12,50 @@
 from ott.solvers.linear.sinkhorn import Sinkhorn
 from pandas import Series
 from rich.progress import track
-from scipy.sparse import issparse
-from scipy.spatial.distance import cosine
+from scipy.sparse import csr_matrix, issparse
+from scipy.spatial.distance import cosine, mahalanobis
 from scipy.special import gammaln
 from scipy.stats import kendalltau, kstest, pearsonr, spearmanr
 from sklearn.linear_model import LogisticRegression
 from sklearn.metrics import pairwise_distances, r2_score
 from sklearn.metrics.pairwise import polynomial_kernel, rbf_kernel
+from sklearn.neighbors import KernelDensity
 from statsmodels.discrete.discrete_model import NegativeBinomialP
 
+
+def compute_medoid(arr, axis=None):
+    if len(arr) == 0:
+        return None  # Handle the case when the array is empty
+
+    if axis is not None:
+        # If axis is specified, compute the medoid along that axis
+        return np.apply_along_axis(lambda x: compute_medoid(x), axis, arr)
+
+    # Calculate pairwise distances between all elements
+    distances = np.abs(arr[:, np.newaxis] - arr)
+
+    # Calculate the total distance for each element
```

Comment:
Very chatGPT like comment. The code is documents this already. Please remove the comment.